cmd/manager/main.go

```diff
@@ -225,3 +252,61 @@ func checkIfRequiredFilesExist(requiredFiles []string) error {
 	}
 	return nil
 }
+
+// validateK8sVersion checks for valid k8s version in the cluster. It returns an error for all versions not equal
+// to supported major version. This is being done this way, and not by directly getting the cluster version, as OpenShift CI
+// returns version in the format 0.0.x and not the actual version attached to its clusters.
+func (c *clusterConfig) validateK8sVersion() error {
+	versionInfo, err := c.oclient.Discovery().ServerVersion()
+	if err != nil {
+		return errors.Wrap(err, "error retrieving server version ")
+	}
+	k8sVersion := versionInfo.GitVersion
+	clusterBaseVersion := k8sVersion[1:strings.LastIndex(k8sVersion, ".")]
+
+	if strings.Compare(clusterBaseVersion, baseK8sVersion) != 0 {
+		return errors.Errorf("Unsupported server version: %v. Supported version is v%v.x", k8sVersion,
+			baseK8sVersion)
+	}
+	return nil
+}
+
+// newClusterConfig creates clusterConfig struct that holds information of the cluster configurations
+func newClusterConfig(config *rest.Config) (*clusterConfig, error) {
```

Comment:
nit: We should get better at organizing code. This should be placed before any of the struct's methods.

Reply:
Done